src/diff.ts

```diff
@@ -55,12 +72,48 @@ export function parseFileDiff(file: File): FileDiff {
     hunks.push(currentHunk);
   }
 
+  hunks = hunks.map((hunk) => {
+    return {
+      ...hunk,
+      commentChains: generateCommentChains(file, hunk, reviewComments),
+    };
+  });
+
   return {
     ...file,
     hunks,
   };
 }
 
+function generateCommentChains(
+  file: File,
+  hunk: Hunk,
+  reviewComments: ReviewComment[]
+): { comments: ReviewComment[] }[] {
+  const topLevelComments = reviewComments.filter(
+    (c) =>
+      !c.in_reply_to_id &&
+      c.path === file.filename &&
+      c.body.length &&
+      c.line &&
+      c.line <= hunk.endLine &&
+      c.line >= hunk.startLine &&
+      (!c.start_line ||
+        (c.start_line <= hunk.endLine && c.start_line >= hunk.startLine))
+  );
```

Comment:
The filter operation assumes `reviewComments` is always an array. Consider adding a null check or default value like `reviewComments ?? []` to handle cases where `reviewComments` might be undefined or null, preventing potential runtime errors.